### ISSUE TITLE
App does not fetch new episodes when there's a future dated episode in the podcast

### ISSUE THREAD

**Issue by reginabally** [Automattic/pocket-casts-ios#724]
Internal discussion: p1676519952301769-slack-C02A333D8LQ
Reported in: 5924448-zd-woothemes

When a podcast has an episode that has a published date set to a future date, the app will not update new episodes unless the user unsubscribes and subscribe to the podcast again.

We observed these events in the logs, where the "DNR LIVE STREAMING SHOW" episode has a published date set to a future date.

```
DNR LIVE STREAMING SHOW
2023-02-11 19:00:00 UTC - Listed published date
2023-02-10 00:03:02 UTC - Feed last modified
2017-12-22 00:33:38 UTC - Added to database (2701409 mins earlier)
```

### Expected behavior

The user expects the app will continue updating new episodes without having to unsubscribe and subscribe again.

### Actual behavior

The app will not show new episodes.

### Steps to reproduce the behavior

1. Subscribe to the private feed mentioned in the Slack thread
2. Note down the latest episode in the app
3. Check the episode listing in the app after a new episode has been published
4. Notice that the app didn't pick up the new episode

##### Reported on iPhone15,3, iOS 16.2, Pocket Casts 7.30

**Comment by emilylaguna** [comment]
Here's some technical info about this issue. 

### tl;dr
The logic to get new episodes is for the apps to tell the server what the latest episode it has and the server returns everything published after that. This means future episodes will always be counted as the latest and the server will never send us any new episodes because it thinks we have them all.

### How do we determine when there are new episodes to be delivered to a device?
The local app database keeps track of the "latest episode" in the `latestEpisodeUuid` column in the SJPodcast table. 

When we perform a sync we want to make sure that data the app has is current, so it [takes each of the subscribed podcasts](https://github.com/Automattic/pocket-casts-ios/blob/33a541ed0af33691a6d1b462dc84a1a1956ad0c0/Modules/Server/Sources/PocketCastsServer/Public/Refresh/MainServerHandler.swift#L231-L233) and updates the above fields with the latest episode from our local episodes database. 

The [logic to find](https://github.com/Automattic/pocket-casts-ios/blob/ff72cd21b886f8b05fd9336093dc46f3e7110c4b/Modules/DataModel/Sources/PocketCastsDataModel/Private/Managers/EpisodeDataManager.swift#L97-L99) the latest episode is this:

> Look at all the episodes for this podcast that we have, order them by the date they were published from newest to oldest, and then give me the first item which should be the latest episode

The [next step is the app](https://github.com/Automattic/pocket-casts-ios/blob/33a541ed0af33691a6d1b462dc84a1a1956ad0c0/Modules/Server/Sources/PocketCastsServer/Public/Refresh/MainServerHandler.swift#L239-L240) send all the subscribed podcasts, and the latest episodes to the server.

The server then looks at this information and does the following to return the latest episodes: 

> Take the latest episode the app has and look it up in my database and retrieve its publish date. Then I'll return every episode for that podcast that is published after that publish date. 

This information is then sent back to the app where it processes the new episodes and adds them to the database. 

#### An example
I am subscribed to a podcast called "Totally Not Real" the last time I sync'd with the server it had 3 episodes:

| Name | Published |
|:---:|:---:|
| Fake Episode 3 | Jan 3rd 1970 |
| Fake Episode 2 | Jan 2nd 1970 |
| Fake Episode 1 | Jan 1st 1970 |

Since the last time I sync'd 2 new episodes have been published and the servers data now looks like this:

| Name | Published |
|:---:|:---:|
| Fake Episode 5 | Jan 5th 1970 |
| Fake Episode 4 | Jan 4th 1970 |
| Fake Episode 3 | Jan 3rd 1970 |
| Fake Episode 2 | Jan 2nd 1970 |
| Fake Episode 1 | Jan 1st 1970 |

When the app syncs it tells the server "Hey the latest episode I have is Fake Episode 3" and the server says "Okay since that was published on Jan 3rd, here are all the episodes since that date which is: Fake Episode 4, and Fake Episode 5"

The app takes that and says "Awesome, I've updated my database"

If I were to do another sync I would send "Fake Episode 5" and the server would say "Nope, that's the latest! You're good"

#### The Problem

If a podcast has a future episode listed the apps episode list now looks like this:

| Name | Published |
|:---:|:---:|
| End of the World Episode | Dec 31st 9999 |
| Fake Episode 3 | Jan 3rd 1970 |
| Fake Episode 2 | Jan 2nd 1970 |
| Fake Episode 1 | Jan 1st 1970 |

Since the logic to get the latest episode is "Sort the episodes by most recently published and give me the first one" we will now _**always**_ return the future episode since it's "the most recently published"

When we sync with the server it will always think we have the latest episode and never return any of the ones published that appear before the future date.

#### A potential solution

When the app looks at the local database for the latest episode, it could ignore any future publish dates and only return episodes that are "from the past". This would make sure the server sends any new episodes published between the past and future episodes.

While this logic change should be safe (famous last words), we'll want to verify and test that it won't cause any other potential issues before deciding.

**Comment by leandroalonso** [comment]
#1251 should have fixed this
